src/wp-includes/rest-api/endpoints/class-wp-rest-template-revisions-controller.php

```diff
@@ -0,0 +1,300 @@
+				'([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)',
+				// Matches the template name.
+				'[\/\w%-]+',
+				$this->rest_base
+			),
+			array(
+				'args'   => array(
+					'parent' => array(
+						'description'       => __( 'The id of a template' ),
+						'type'              => 'string',
+						'sanitize_callback' => array( $this->parent_controller, '_sanitize_template_id' ),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			sprintf(
+				'/%s/(?P<parent>%s%s)/%s/%s',
+				$this->parent_base,
+				/*
+				 * Matches theme's directory: `/themes/<subdirectory>/<theme>/` or `/themes/<theme>/`.
+				 * Excludes invalid directory name characters: `/:<>*?"|`.
+				 */
+				'([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)',
+				// Matches the template name.
+				'[\/\w%-]+',
+				$this->rest_base,
+				'(?P<id>[\d]+)'
+			),
+			array(
+				'args'   => array(
+					'parent' => array(
+						'description'       => __( 'The id of a template' ),
+						'type'              => 'string',
+						'sanitize_callback' => array( $this->parent_controller, '_sanitize_template_id' ),
+					),
+					'id'     => array(
+						'description' => __( 'Unique identifier for the revision.' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+					'args'                => array(
+						'force' => array(
+							'type'        => 'boolean',
+							'default'     => false,
+							'description' => __( 'Required to be true, as revisions do not support trashing.' ),
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Get the parent post, if the ID is valid.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param int $parent_post_id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_parent( $parent_post_id ) {
+		$error = new WP_Error(
+			'rest_post_invalid_parent',
+			__( 'Invalid template parent ID.' ),
+			array( 'status' => 404 )
+		);
+
+		$template = get_block_template( $parent_post_id, $this->parent_post_type );
+
+		if ( ! $template ) {
+			return $error;
```

Comment:
```suggestion
                         $template = get_block_file_template( $request['id'], $this->post_type );
                         if( $template ){
                            return new WP_Error(
			          'rest_post_invalid_parent_type',
			          __( 'Invalid template type, template must be saved as post.' ),
			         array( 'status' => 404 )
		);
			return $error;
```

I keep wonder if we need to do this?

Reply:
Well spotted. This also alarmed me.
Fixed in e852cc603fca5cd230467f928f32a59c790d03d0.